master/static/migrations/20240321174101_handle-single-run-experiments-best-trial.tx.up.sql

```diff
@@ -0,0 +1,52 @@
+CREATE OR REPLACE FUNCTION autoupdate_exp_best_trial_single_run_no_validation() RETURNS trigger AS $$
+BEGIN 
+    WITH rc AS (
+        SELECT COUNT(*) AS count
+        FROM runs
+        WHERE experiment_id = NEW.id),
+    br AS (
+        SELECT id, best_validation_id 
+        FROM runs 
+        WHERE 
+            runs.experiment_id = NEW.id 
+            AND
+            best_validation_id IS NULL
+        ORDER BY searcher_metric_value_signed LIMIT 1)
+    UPDATE 
+        experiments 
+    SET 
+        best_trial_id = br.id
+    FROM br, rc
+    WHERE
+        experiments.id = NEW.id
+        AND
+        rc.count = 1;
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS autoupdate_exp_best_trial_single_run_no_validation_trigger ON experiments;
+CREATE TRIGGER autoupdate_exp_best_trial_single_run_no_validation_trigger 
+AFTER UPDATE OF state ON experiments
+FOR EACH ROW 
+    WHEN (NEW.state = 'COMPLETED' AND NEW.best_trial_id IS NULL)
+    EXECUTE PROCEDURE autoupdate_exp_best_trial_single_run_no_validation();
+
+-- backfill existing single-trial experiments without best_trial_id
+WITH single_run_experiments AS (
+    SELECT experiment_id
+    FROM runs r
+    GROUP BY experiment_id
+    HAVING COUNT(*) = 1),
```

Comment:
I'm kinda split on this

What do you think about not checking the count of the experiment, and instead checking the searcher type. I think the searcher type check is simpler and I don't know how much this case would come up in other searcher types

Reply:
Would we be concerned about custom searchers with single runs? I would under the impression that a single-run custom searcher would be a misuse of custom search.

Reply:
My understanding is no customer has ever used custom searcher and we want to remove it and replace it with generic tasks